metrics-monitoring/README.adoc

```diff
@@ -0,0 +1,225 @@
+////
+START GENERATED ATTRIBUTES
+WARNING: This content is generated by running npm --prefix .build run generate:attributes
+////
+
+
+:org-name: Application Services
+:community:
+:imagesdir: ./images
+:product-version: 1
+:product-long: OpenShift Streams for Apache Kafka
+:product: Streams for Apache Kafka
+:registry-product-long: OpenShift Service Registry
+:registry: Service Registry
+// Placeholder URL, when we get a HOST UI for the service we can put it here properly
+:service-url: https://console.redhat.com/application-services/streams/
+:registry-url: https://console.redhat.com/application-services/service-registry/
+:property-file-name: app-services.properties
+
+// Other upstream project names
+:samples-git-repo: https://github.com/redhat-developer/app-services-guides
+
+//URL components for cross refs
+:base-url: https://github.com/redhat-developer/app-services-guides/blob/main/
+:base-url-cli: https://github.com/redhat-developer/app-services-cli/tree/main/docs/
+:getting-started-url: getting-started/README.adoc
+:getting-started-service-registry-url: getting-started-service-registry/README.adoc
+:kafka-bin-scripts-url: kafka-bin-scripts/README.adoc
+:kafkacat-url: kafkacat/README.adoc
+:quarkus-url: quarkus/README.adoc
+:quarkus-service-registry-url: quarkus-service-registry/README.adoc
+:rhoas-cli-url: rhoas-cli/README.adoc
+:rhoas-cli-kafka-url: rhoas-cli-kafka/README.adoc
+:rhoas-cli-service-registry-url: rhoas-cli-service-registry/README.adoc
+:rhoas-cli-ref-url: commands
+:topic-config-url: topic-configuration/README.adoc
+:consumer-config-url: consumer-configuration/README.adoc
+:service-binding-url: service-discovery/README.adoc
+:access-mgmt-url: access-mgmt/README.adoc
+:access-mgmt-service-registry-url: access-mgmt-service-registry/README.adoc
+
+////
+END GENERATED ATTRIBUTES
+////
+
+[id="chap-monitoring-metrics"]
+= Monitoring metrics in {product-long}
+ifdef::context[:parent-context: {context}]
+:context: monitoring-metrics
+
+// Purpose statement for the assembly
+[role="_abstract"]
+As a developer or administrator, you can view metrics in {product} to visualize the performance and data usage for Kafka instances and topics that you have access to. You can view metrics directly in the {product} web console, or use the metrics API endpoint provided by {product} to import the data into your own metrics monitoring tool, such as Prometheus.
+
+//Additional line break to resolve mod docs generation error, not sure why. Leaving for now. (Stetson, 20 May 2021)
+
+[id="ref-supported-metrics_{context}"]
+== Supported metrics in {product}
+
+{product} supports the following metrics for Kafka instances and topics. In the {product} web console, a subset of these metrics is displayed in the *Dashboard* page of each Kafka instance.
+
+Broker metrics::
```

Comment:
Do we need to describe the labels each of these metrics will have? I think it would make it a little clearer for some of the metrics, like `kafka_topic:kafka_topic_partitions:count` (where I assume there's a series for each topic).

Reply:
That would be nice. Reason we've just listed them is that this is what the user will have to choose from when they set them up initially. Then when they actually appear, they new label is shown, which I agree might not always be super intuitive like you say. If those labels are pretty certain, I agree would be nice to list them, like in parentheses or something for each. But I'd need some help with that yet again. Just getting this list of metrics required help from Pete.

Reply:
Labels should be constant, they're defined [here](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/constants/metrics.go). The `VariableLabels` contains the labels returned for each metric.

Reply:
@JameelB ack, thanks.